Default to the new UI for first time users

whymarrh opened the pull request:
Closes #4971

This PR is the start of the new UI switch, showing the new UI by default for first time users.

~~We won't be merging this PR for a while, FYI.~~ This PR is ready for review.

Changes:

- [x] Remove instances of "beta" from the old&new UIs
- [x] Hide the main announcement for users actively switching back to the old UI
- [x] Update end-to-end tests to reflect new defaults

whymarrh committed:
Default to the new UI for first time users

whymarrh committed:
Remove "Beta" from header logo

whymarrh committed:
Update new UI links in the old UI s/Beta/New/

whymarrh committed:
Don't announce new UI to users switching back

whymarrh committed:
Update new UI e2e tests for new UI default

whymarrh commented:
I'm torn about what we should do with the old UI e2e tests here as updating the e2e tests would require going through the new UI on-boarding to be able to switch back to the old UI. We could delete the old UI e2e tests? It's a non-zero risk though.<sup>(Is it really? They're pretty flaky anyhow tbf.)</sup>

Similar story for the old UI integration tests. Maybe this PR represents the end of the road for all old UI tests?

tmashuang commented:
All for deleting the old oldUi e2e tests and integration tests. For the new Ui.

1blockologist commented on app/scripts/controllers/preferences.js:
```diff
@@ -32,7 +32,9 @@ class PreferencesController {
       useBlockie: false,
-      featureFlags: {},
+      featureFlags: {
+        betaUI: true,
```
kind of ironic to push people toward the new UI and still call it betaUI here

whymarrh replied:
What do you mean?

1blockologist replied:
the flag is called "betaUI" and is references throughout the pull request.

when the reality is that it will become the production UI, by nature of being the primary one people get to see and is the natural progression of it.

the idea being that some OTHER unforeseen design become the new betaUI, or the community itself is leveraged to provide skins for metamask

whymarrh replied:
I don't follow, sorry. This is a feature flag that is now permanently enabled as the first step to us removing its usages. It is called `betaUI` because that's what the feature it _was_. We will be removing it because, yes, it is becoming the default UI. I don't think that's ironic/unexpected, that's the progression of a feature flag.

If you're interested in community skins, feel free open up a new issue detailing those ideas!

danfinlay replied:
I get it, that's funny. With hindsight, we could have named this parameter something like "ui2" instead. It's like finding an old can of soap in your basement that says "new improved formula!". How new? How beta?

Not a big deal, we can just remove this flag when it's no longer a beta.

whymarrh commented:
Looks like we'll need to remove update our GH branch settings if we're gonna remove the old UI tests:

<img width="758" src="https://user-images.githubusercontent.com/1623628/46155037-fd70cb00-c250-11e8-8f5f-ac64a5cfbe85.png">

tmashuang commented:
https://github.com/MetaMask/metamask-extension/blob/538b91c4abd865751e92351d58fa11a74fda73c4/.circleci/config.yml#L151

tmashuang commented on package.json:
```diff
@@ -13,8 +13,6 @@
     "watch:test:unit": "nodemon --exec \"npm run test:unit\" ./test ./app ./ui",
     "test:unit": "cross-env METAMASK_ENV=test mocha --exit --require test/setup.js --recursive \"test/unit/**/*.js\" \"ui/app/**/*.test.js\" && dot-only-hunter",
     "test:single": "cross-env METAMASK_ENV=test mocha --require test/helper.js",
-    "test:integration": "npm run test:integration:build && npm run test:flat && npm run test:mascara",
```
The issue with this is that the `test-integration-flat-firefox/chrome` is also used for `test-integration-mascara`

tmashuang commented on package.json:
```diff
@@ -13,8 +13,6 @@
     "test:unit": "cross-env METAMASK_ENV=test mocha --exit --require test/setup.js --recursive \"test/unit/**/*.js\" \"ui/app/**/*.test.js\" && dot-only-hunter",
     "test:single": "cross-env METAMASK_ENV=test mocha --require test/helper.js",
-    "test:integration": "npm run test:integration:build && npm run test:flat && npm run test:mascara",
-    "test:integration:build": "gulp build:scss",
```
The issue with this is that the `test-integration-flat-firefox/chrome` is also used for `test-integration-mascara`

danfinlay commented:
Interesting Q about the test suite. It's *mostly* fine to remove old UI tests, except that we could catastrophically break it, and we probably wouldn't notice, since manually testing is just as tedious as it automatically clicking through just to enable the old UI.

That said, the things we are changing right now are just background + new UI, and anything in background that would break the old UI would probably break the new UI too, so maybe we can just be very mindful about breaking background API changes during this interim?

Lastly, how hard would it be for us to copy the new-ui setup to the beginning of the old-UI tests, so we can get the app to a state where they make sense? @tmashuang could you briefly review those files and see if they might compose at all? The old UI tests will probably need some of their first time flow tests removed, since it is no longer exactly possible to encounter.

whymarrh commented:
> Interesting Q about the test suite. It's mostly fine to remove old UI tests, except that we could catastrophically break it, and we probably wouldn't notice

That's a risk, certainly. I would argue that the risk is small though as a lot of functionality/actions are shared and breaking changes would likely be clear during review. I think it's possible to go a single release cycle without old UI e2e tests and without regressions.

kumavis commented:
Maybe we can run a command from the test to switch to old ui even though there is not a ui option to do it

whymarrh commented:
@kumavis to follow-up to my suggestions above, I'm thinking we can remove `test-integration-flat-chrome` and `test-integration-flat-firefox` from the set of required checks for PRs. I don't think it's worth hacking the old UI tests just to get a green check. The plan is to have the old UI remove completely the release after the one this is is so we're talking about a single release without test coverage for the old UI (which has historically not been a particularly good indicator of anything).

whymarrh committed:
Remove "Beta" from welcome messages

 More sed magic:

    sed -i.bak 's/MetaMask Beta/MetaMask/' app/_locales/**/messages.json && \
    find app/_locales -name '*.bak' -delete

    sed -i.bak 's/"welcomeBeta"/"welcome"/' app/_locales/**/messages.json && \
    find app/_locales -name '*.bak' -delete

whymarrh committed:
Remove Mascara from CircleCI config

whymarrh committed:
ci: Drop old UI e2e tests from CircleCI

whymarrh committed:
Speed up genStates script by using built-in require fn

whymarrh committed:
Remove first-time and navigate-txs integration tests

whymarrh committed:
Add timeout to waitUntilXWindowHandles e2e helper fn

danjm commented on test/integration/index.js:
```diff
@@ -2,7 +2,7 @@ const fs = require('fs')
 const pump = require('pump')
 const browserify = require('browserify')
-const tests = fs.readdirSync(path.join(__dirname, 'lib'))
+const tests = fs.readdirSync(path.join(__dirname, 'lib')) // .filter((filename) => filename.match(/send/))
```
delete this change

whymarrh replied:
Thanks 😅

danfinlay reviewed (approved):
I'm for it. It's very much time!
